[MemoryDataSource] Add optional 'deleteData' parameter to constructor to allow management of 'data' parameter lifespan outside of libtheoraplayer

jleveque opened the pull request:
Allow more flexibility for managing lifespan of `data` parameter outside of libtheoraplayer.

This is helpful in the following situations:
  - Variable passed as `data` needs to live longer than the MemoryDataSource created from it.
  - Variable passed as `data` was allocated using a `malloc` family function.
    - In the destructor, `data` is freed using `delete[]`, which assumes it was allocated using `new[]`. This creates a mismatch if the `data` parameter was allocated using a function from the `malloc` family.

The new `deleteData` parameter defaults to `true` so as not to have an impact on any internal theoraplayer source code. It must be explicitly set to `false` when calling the constructor.

kspes commented:
I agree, it makes perfect sense, can't believe I didn't think of this case :)

Just adjust the code to match the library convention:
* deleteData = true (space  between operators)
* tabs instead of spaces for indentation, you can see in the git commit diff it doesn't match.
when set, I'll merge the pull request.

jleveque commented:
Done. I made sure all the line I added was indented with tabs to be consistent with your style, and I also fixed line 27 of the cpp file which was already incorrect in the process, but I missed line 21 which was also already incorrect.

With this new commit, I replaced all remaining 4-space indentations in these files with tabs (8 additional lines).

kspes commented:
all is good now, merged! Thanks for your contribution :)
